src/server.ts

```diff
@@ -0,0 +1,16 @@
+import http from 'http';
+
+import dotenv from 'dotenv';
+dotenv.config();
+
+import api from './api';
+import { connectMONGODB} from './services/mongodb';
+
+const server = http.createServer(api);
```

Comment:
Why are you using `http`?

Reply:
do you mean why not https or why http generally?

Reply:
why http generally, why not initialize it with express?

Reply:
It is for separation between the logic related to the `API` from the logic related to the `Server` 
for example:
- Connect to the database is a concern of server not the api 
- Handle routing is a concern of api not the server

Reply:
I don't mean the separation, I mean creating the server from `http` and not `express`.

Like why using http to create the server
```
const server = http.createServer(api);
```
instead of creating it with express
```
const server = express();
server .use(api);
```

Reply:
there is no difference between both except that we will use the same syntax 
```
   const app = express();
   
   const options = {
    key: fs.readFileSync('key.pem'),
    cert: fs.readFileSync('cert.pem')
   };
  
   const server = https.createServer(options, app);
```
when we try to use a TLS certificate to enable https.
note that key.pem and cert.pem are the certificate files obtained from the CA